From 24f14fc16f5ef9046987d45019d98523c27c506c Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh+mozilla@glandium.org>
Date: Tue, 9 Jun 2015 09:54:24 +0900
Subject: [PATCH] Bug 1172632 - Don't guess malloc_usable_size type of argument
 based on ANDROID_VERSION in mozjemalloc, but use the result of the configure
 test instead

---
 memory/mozjemalloc/jemalloc.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/memory/mozjemalloc/jemalloc.c b/memory/mozjemalloc/jemalloc.c
index f8e36e1..9113d26 100644
--- a/memory/mozjemalloc/jemalloc.c
+++ b/memory/mozjemalloc/jemalloc.c
@@ -6509,23 +6509,18 @@ malloc_good_size_impl(size_t size)
 		 * malloc_good_size(n).
 		 */
 		size = PAGE_CEILING(size);
 	}
 	return size;
 }
 
 
-#if defined(MOZ_MEMORY_ANDROID) && (ANDROID_VERSION < 19)
 MOZ_MEMORY_API size_t
-malloc_usable_size_impl(void *ptr)
-#else
-MOZ_MEMORY_API size_t
-malloc_usable_size_impl(const void *ptr)
-#endif
+malloc_usable_size_impl(MALLOC_USABLE_SIZE_CONST_PTR void *ptr)
 {
 	DARWIN_ONLY(return (szone->size)(szone, ptr));
 
 #ifdef MALLOC_VALIDATE
 	return (isalloc_validate(ptr));
 #else
 	assert(ptr != NULL);
 
-- 
2.4.0.2.g36460d1.dirty

